### DISCUSSION TITLE
7.0.0 (2024-12-03)

### DISCUSSION THREAD

**Discussion by manusa** [fabric8io/kubernetes-client#6697] [Announcements]
#### Bugs
- Fix #6038: Support for Gradle configuration cache
- Fix #6214: Java generator does not recognize fields in CRDs other than metadata, spec, and status
- Fix #6459: Pod log request sinceTime param correctly encoded
- Fix #6632: Mock server creationTimestamp and deletionTimestamp formatted consistently (ISO 8601)
- Fix #6654: (crd-generator) default values for CRD fields can be numeric or boolean

#### Improvements
- Fix #3069: (crd-generator) Add `@AdditionalPrinterColumn` to specify a printer column by JSON path.
- Fix #6392: (crd-generator) Add `@AdditionalSelectableField` and `@SelectableField` to specify selectable fields.
- Fix #5836: (crd-generator) Add `@Size` annotation to limit the size of strings, lists/arrays or maps
- Fix #5868: (crd-generator) Add `exlusiveMinimum` / `exclusiveMaximum` support to `@Min` and `@Max`
- Fix #5264: Remove deprecated `Config.errorMessages` field
- Fix #6008: removing the optional dependency on bouncy castle
- Fix #6407: sundrio builder-annotations is not available via bom import
- Fix #6230: introduced Quantity.multiply(int) to allow for Quantity multiplication by an integer
- Fix #6281: use GitHub binary repo for Kube API Tests
- Fix #6282: Allow annotated types with Pattern, Min, and Max with Lists and Maps and CRD generation
- Fix #5480: Move `io.fabric8:zjsonpatch` to KubernetesClient project
- Fix #6240: Support for multiple files listed in the KUBECONFIG env var
- Fix #6655: Support removing ResourceEventHandler for informers

#### Dependency Upgrade
- Fix #2632: Bumped OkHttp from 3.12.12 to 4.12.0
- Fix #5965: Bumped sundrio.version from 0.103.1 to 0.200.0
- Fix #6052: Removed dependency on no longer maintained com.github.mifmif:generex
- Fix #6676: Removed tests and examples from the BOM

#### New Features
- Fix #5719: io.fabric8:mockwebserver is now based on Vert.x
- Fix #5795: (crd-generator) Allow to configure categories
- Fix #5944: (crd-generator) Add CRD-Generator Maven Plugin
- Fix #5958: (crd-generator) Add CRD-Generator CLI Application
- Fix #6145: Introduced `io.fabric8:kubernetes-model-validator` module to validate Kubernetes resources
- Fix #6150: Config uses `proxy-url` in kubeconfig's cluster configuration

#### _**Note**_: Breaking changes
- Check detailed migration documentation for breaking changes in [7.0.0](./doc/MIGRATION-v7.md)
- Fix #4659: Removed `SupportTestingClient` interface
- Fix #5264: `Config.errorMessages` has been removed. Please use Kubernetes status messages directly.
- Fix #5351: Removed deprecated `io.fabric8:openshift-server-mock` artifact
- Fix #6081: Moved Java baseline from 8 (1.8) to 11
- Fix #6138: Removed unused `io:fabric8:kubernetes-model` artifact
- Fix #6140: withReadyWaitTimeout defaults to 0, which means no waiting
- Fix #6156: Removed deprecated extension `io:fabric8:service-catalog`
- Fix #6158: Removed deprecated methods from `io.fabric8.kubernetes.client.utils.IOHelpers` class
- Fix #6159: Removed deprecated `io.fabric8.kubernetes.client.utils.Utils.getPluralFromKind` method
- Fix #6361: Renamed SettableBeanPropertyDelegate to SettableBeanPropertyDelegating
- Fix #6470: Switched default HTTP client from OkHttp to Vert.x (`kubernetes-httpclient-vertx`)
- Fix #6603: Removed deprecated `io.fabric8.openshift.api.model.runtime.RawExtension` class
- Fix #6605: Removed deprecated `ApiVersionUtil` classes in extension modules
- Fix #6609: Removed deprecated `io.fabric8.crd.generator.CRDInfo.getVersion` method
- Fix #6612: Removed deprecated `io.fabric8.kubernetes.client.Client.isAdaptable` method
- Fix #6626: Removed deprecated methods from `io.fabric8.kubernetes.client.osgi.ManagedKubernetesClient` class
- Fix #6627: Removed deprecated methods from `io.fabric8.kubernetes.client.CustomResource` class
- Fix #6628: Removed deprecated `io.fabric8.kubernetes.client.CustomResourceList` class
- Fix #6630: Removed deprecated `io.fabric8.kubernetes.client.utils.ReflectUtils` class
- Fix #6455: (crd-generator) Changed format in `@PrinterColumn` to enum
- Fix #6673: (crd-generator) Deprecation of CRD Generator v1
- Fix #6676: Removed tests and examples from the BOM

## New Contributors
* @Sineaggi made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6038
* @fhussonnois made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6213
* @fxshlein made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6343
* @aswinayyolath made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6615
* @currantino made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6634
* @asharan2buff made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6633
* @emmanuel-ferdman made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6653
* @MichaelMorrisEst made their first contribution in https://github.com/fabric8io/kubernetes-client/pull/6457

**Full Changelog**: https://github.com/fabric8io/kubernetes-client/compare/v6.13.4...v7.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fabric8io/kubernetes-client/releases/tag/v7.0.0'>7.0.0 (2024-12-03)</a>.</em>

**Comment by scholzj** [comment]
Thanks for the release.

Are there any plans to add support for the new Kubernetes APIs from Kube 1.31? For example the [ImageVolumeSource](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#imagevolumesource-v1-core)? It does not seem to be supported in 7.0.0 (or at least I did not found it).

**Reply by manusa** [reply]
Will be added in v7.1.0 (https://github.com/fabric8io/kubernetes-client/issues/5993) which should not take long.

I wanted to have the initial v7 release with no model incompatibilities.

_n.b. In case someone else is reading this, note that v7.0.0 (and any v6.x version) is fully compatible with Kubernetes v1.31. The only thing is that types and builders are not provided for v1.31+ resources._

**Reply by scholzj** [reply]
Great, thanks. I will be looking forward to 7.1.0. 7.0.0 seems to work fine so far in my tests with Strimzi and Kube 1.31 - we will use it in the mean-time. It was just the new APIs that I was wondering about.

**Reply by manusa** [reply]
Great news. Thanks for sharing, please keep us posted in case something comes up.

**Comment by anton-subbotin** [comment]
Hello, thank you for the release.

As a part of https://github.com/fabric8io/kubernetes-client/issues/6421 , `expressionSelectors` support was removed from `PodSelectorSpec` in chaos-mesh schema. It breaks the chaos-mesh client usage, and is not listed under the breaking changes.

Are there any plans to address this?

**Reply by manusa** [reply]
This isn't included because Chaos Mesh has the field marked as `swaggerignore:true`, hence not exposing to to the OpenAPI spec:

https://github.com/chaos-mesh/chaos-mesh/blob/f3d3540bbe2e479ad1516791b046bc19fe46bf6e/api/v1alpha1/selector.go#L60

You might want to open an issue on the Chaos Mesh repository to ask why is this field not exposed.

**Reply by anton-subbotin** [reply]
Thanks @manusa for the prompt response!

**Reply by manusa** [reply]
No problem, please link any discussion or issue you open with Chaos Mesh so we get awareness.

**Reply by anton-subbotin** [reply]
Discussion with Chaos Mesh: https://github.com/chaos-mesh/chaos-mesh/discussions/4588

**Reply by manusa** [reply]
https://github.com/chaos-mesh/chaos-mesh/pull/4684

**Comment by alexrosenfeld10** [comment]
Where did the `setPathType` method on the `io.fabric8.kubernetes.api.model.extensions.HTTPIngressPath` class go in the 7.0.0 release?

**Reply by manusa** [reply]
I don't think that class has ever had a `pathType` field.
You're probably talking about the class in the `networking.k8s.io/v1` apiVersion.

https://github.com/fabric8io/kubernetes-client/blob/c5f6674093d8cf7c4599c1c5dbdfa8ae32e6f12a/kubernetes-model-generator/kubernetes-model-networking/src/generated/java/io/fabric8/kubernetes/api/model/networking/v1/HTTPIngressPath.java#L116-L122
